File: nodejs/muse/algorithms/sorting/InsertionSortTest.js
"use strict"

const InsertionSort   = require("muse/algorithms/sorting/InsertionSort")
const SequenceBuilder = require("muse/util/SequenceBuilder")
const Sequences       = require("muse/util/Sequences")
const TestRunner      = require("muse/util/TestRunner")

const testInsertionSort = () => {
    let size = 32768

    const arr = new Array(size)
    SequenceBuilder.packRandom(arr)

    let checksum = Sequences.parityChecksum(arr)

    InsertionSort.sort(arr)

    if (Sequences.parityChecksum(arr) !== checksum) {
        return false
    }

    return Sequences.isSorted(arr)
}

if (module === require.main) {
    TestRunner.pick(testInsertionSort)
}

File: java/muse/algorithms/sorting/InsertionSortTest.java
import muse.algorithms.sorting.InsertionSort;
import muse.util.SequenceBuilder;
import muse.util.Sequences;
import muse.util.TestRunner;

public class InsertionSortTest {

  public static boolean testInsertionSort() {
    int size = 32768;

    Integer[] arr = new Integer[size];
    SequenceBuilder.packRandom(arr);

    int checksum = Sequences.parityChecksum(arr);

    InsertionSort.sort(arr);

    if (Sequences.parityChecksum(arr) != checksum) {
      return false;
    }

    return Sequences.isSorted(arr);
  }

  public static void main(String[] args) {
    TestRunner.pick(InsertionSortTest::testInsertionSort);
  }
}
